script/build.js

```diff
@@ -136,6 +137,28 @@ smith.use(ignore(ignoreList));
 // plugin chain.
 smith.use(filenames());
 
+smith.use((files, metalsmith, done) => {
+  runningCount = 0;
+  Object.keys(files).forEach((filename) => {
+    if (filename.match(/\.md$/) !== null) {
+      const fullPath = "content/pages/" + filename;
+      if (fs.existsSync(fullPath)) {
+        runningCount += 1;
+        git.log({ file: fullPath }, function(err, log) {
+          if (log.all.length > 1 && files[filename]) {
+            files[filename].lastupdate = log.latest.date;
+          }
+        }).then(function() {
+          runningCount -= 1;
+          if (runningCount == 0) {
+            done();
+          }
+        });
+      }
+    }
+  });
+});
+
```

Comment:
Few notes:

* It's annoying that the file paths don't include the full path, so I have to make the assumption that all markdown content we want to deal with is in `content/pages` and that seems crappy.
* The log listing is async, and I feel like the thing with the running count is super cheesy. Is there a better way to do this?